File: Dart/length-of-last-word.dart
class Solution {
  int lengthOfLastWord(String s) {
    int result = 0;
    int i = s.length - 1;
    while ( 0 <= i&&s[i] == ' ' ) i--;
    while (0 <= i&&s[i] != ' ' ) {
      result++;
      i--;
    }
    return result;
  }
}

File: Python/length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        len_s = len(s)
        i = len_s - 1
        while i >= 0 and s[i] == " ":
            i -= 1
        count = 0
        while i >= 0 and s[i] != " ":
            i -= 1
            count += 1
        return count


if __name__ == "__main__":
    sol = Solution()
    test_str = "Hello World"
    print(sol.lengthOfLastWord(test_str))
